test/test_optimizers/test_sgd.py

```diff
@@ -0,0 +1,42 @@
+from yadll.tensor.autodiff import *
+from yadll.tensor.optimizers import *
+from yadll.tensor.nn import *
+import numpy
+import torch
+
+
+# TODO Add some unit-tests for stochastic gradient descent.
+def test_sgd_step_no_momentum():
+    x = Tensor.random((2, 2))
+    torch_x = torch.tensor(x.data)
+    model = Linear(2, 1, False)
+    torch_model = torch.nn.Linear(2, 1, False)
+    torch_model.weight = torch.nn.Parameter(torch.tensor(model.weight.data))
+    optim = SGD(model.parameters(), 0.1, 0)
+    torch_optim = torch.optim.SGD(torch_model.parameters(), 0.1, 0)
+    out = model(x).sum()
+    torch_out = torch_model(torch_x).sum()
+    out.backward()
+    torch_out.backward()
+    assert np.all(model.weight.data == torch_model.weight.detach().numpy()), "weight not equal before step"
+    optim.step()
+    torch_optim.step()
+    assert np.all(model.weight.data == torch_model.weight.detach().numpy()), "weight not equal after step"
+
+
+def test_sgd_with_momentum():
+    x = Tensor.random((2, 2))
+    torch_x = torch.tensor(x.data)
+    model = Linear(2, 1, False)
+    torch_model = torch.nn.Linear(2, 1, False)
+    torch_model.weight = torch.nn.Parameter(torch.tensor(model.weight.data))
+    optim = SGD(model.parameters(), 0.1, 0.9)
+    torch_optim = torch.optim.SGD(torch_model.parameters(), 0.1, 0.9)
```

Comment:
Given that we are changing only the momentum value between the two tests, we could potentially use [pytest.parametrize](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html) to remove some lines. 
I understand that we are testing two different things, but it could suffice to simply mention as a comment that when the last arg = 0, it means there is no momentum.

Reply:
for some tests yeah, but with the recent push they're too different